mllib/src/main/scala/org/apache/spark/ml/feature/Tokenizer.scala

```diff
@@ -39,3 +39,66 @@ class Tokenizer extends UnaryTransformer[String, Seq[String], Tokenizer] {
 
   override protected def outputDataType: DataType = new ArrayType(StringType, false)
 }
+
+
+/**
+ * :: AlphaComponent ::
+ * A regex based tokenizer that extracts tokens using a regex.
+ * Optional additional parameters include enabling lowercase stabdarization, a minimum character
+ * size for tokens as well as an array of stop words to remove from the results.
+ */
+@AlphaComponent
+class RegexTokenizer extends UnaryTransformer[String, Seq[String], RegexTokenizer] {
+
+  val lowerCase = new BooleanParam(this, 
+      "lowerCase", 
+      "enable case folding to lower case", 
+      Some(true))
+  def setLowercase(value: Boolean) = set(lowerCase, value)
+  def getLowercase: Boolean = get(lowerCase)
+
+  val minLength = new IntParam(this, 
```

Comment:
Add doc and update code style. What's the case when we match a token with regex but its length is zero? Should we control it in the regex, e.g., `\d{5,}`?

Btw, it is not intuitive that the min value is excluded. Could we remove "excluded" and set the default value to `1`? And it might be better to call it `minTokenLength`, if we want to keep it.